skopt/space.py

```diff
@@ -0,0 +1,432 @@
+
+    def transform(self, X):
+        """Transform samples form the original space to a warped space."""
+        return self.transformer.transform(X)
+
+    def inverse_transform(self, Xt):
+        """Inverse transform samples from the warped space back into the
+           original space.
+        """
+        return self.transformer.inverse_transform(Xt)
+
+    @property
+    def size(self):
+        return 1
+
+    @property
+    def transformed_size(self):
+        return 1
+
+    @property
+    def bounds(self):
+        raise NotImplementedError
+
+    @property
+    def transformed_bounds(self):
+        raise NotImplementedError
+
+
+class Real(Dimension):
+    def __init__(self, low, high, prior="uniform"):
+        """Search space dimension that can take on any real value.
+
+        Parameters
+        ----------
+        * `low` [float]:
+            Lower bound (inclusive).
+
+        * `high` [float]:
+            Upper bound (exclusive).
+
+        * `prior` ["uniform" or "log-uniform", default="uniform"]:
+            Distribution to use when sampling random points for this dimension.
+            - If `"uniform"`, points are sampled uniformly between the lower
+              and upper bounds.
+            - If `"log-uniform"`, points are sampled uniformly between
+              `log10(lower)` and `log10(upper)`.`
+        """
+        self._low = low
+        self._high = high
+        self.prior = prior
+
+        if prior == "uniform":
+            self._rvs = uniform(self._low, self._high - self._low)
+            self.transformer = _Identity()
+
+        elif prior == "log-uniform":
+            self._rvs = uniform(
+                np.log10(self._low),
+                np.log10(self._high) - np.log10(self._low))
+            self.transformer = _Log10()
+
+        else:
+            raise ValueError(
+                "Prior should be either 'uniform' or 'log-uniform', "
+                "got '%s'." % self._rvs)
+
+    @property
+    def bounds(self):
+        return (self._low, self._high)
+
+    @property
+    def transformed_bounds(self):
+        if self.prior == "uniform":
+            return (self._low, self._high)
+
+        else:  # self.prior == "log-uniform"
+            return (np.log10(self._low), np.log10(self._high))
+
+
+class Integer(Dimension):
+    def __init__(self, low, high):
+        """Search space dimension that can take on integer values.
+
+        Parameters
+        ----------
+        * `low` [float]:
+            Lower bound (inclusive).
+
+        * `high` [float]:
+            Upper bound (inclusive).
+        """
+        self._low = low
+        self._high = high
+        self._rvs = randint(self._low, self._high + 1)
+        self.transformer = _Identity()
+
+    def inverse_transform(self, Xt):
+        """Inverse transform samples from the warped space back into the
+           orignal space.
+        """
+        return super(Integer, self).inverse_transform(Xt).astype(np.int)
```

Comment:
I can't work out why we need this, can you add a comment that explains it to future us?